### PR TITLE
CR-1142 Individual Paper Form

### PR DESCRIPTION
**Pull request by Paul-Joel**
# Motivation and Context
Implements individual requests for paper-form journey, and revises implementation of individual UAC journey

# What has changed
Introduces new 'household interstitial' pages after 'Confirm Address' if addressType is household or SPG. New page explains about household journey, and allows switch to individual journey. Replaces the link that was on the 'Select method' screen.
Introduces new session attribute for 'individual'. This now drives whether individual messaging is shown rather than by url. New attribute is set to 'false' unless starting from the individual UAC journey, and is set to 'true' if individual is selected via the interstitial pages, if a user picks a CE-U address, or if the user declares as a 'resident' on the CE-E Manager or Resident page.
New interstitial page in paper form journey now allows for request of individual paper form/large print form
Full set of unit tests added/updated for new functionality

Requires a matching RHCucumber update (adds the new interstitial page to the test steps)
https://github.com/ONSdigital/census-rh-cucumber/pull/104

# How to test?
Retest of existing UAC individual journey (triggered from /start/ or by selecting individual option via interstitial page in household journey
Test switching to individual journey for paper form via interstitial page in /en/requests/paper-form/enter-address/

# Links
https://github.com/ONSdigital/census-rh-cucumber/pull/104
https://collaborate2.ons.gov.uk/jira/browse/CR-1344
https://collaborate2.ons.gov.uk/jira/browse/CR-1142

# Screenshots (if appropriate):

### CODE REVIEW

File: app/common_handlers.py [hunk 1]
```diff
@@ -219,6 +227,23 @@ async def get(self, request):
...
+        elif user_journey == 'requests':
+            try:
+                individual = session['attributes']['individual']
+            except KeyError:
+                individual = False
+                attributes = {'individual': False}
+                session['attributes'] = attributes
+                session.changed()
+        else:
+            individual = False
```
Comment (rpcatling) [inline app/common_handlers.py:245]: no code coverage for this. Can we get here?

**Review by rpcatling — CHANGES_REQUESTED** [review]
just one minor thing

**Review by rpcatling — APPROVED** [review]

File: app/templates/request-code-select-method.html [hunk 2]
```diff
@@ -60,35 +60,6 @@
...
-                {%- endautoescape -%}
-                {%- call onsPanel({
-                        'type': 'warn'
-                    })
-                -%}
-                <p><strong>{{ _('Someone in your household must still complete a census using a household access code')}}</strong></p>
-                {%- endcall -%}
-            {%- endset -%}
-
-            {%- call onsCollapsible({
```
Comment (MartinHumphrey) [inline app/templates/request-code-select-method.html:77]: Can remove import of onsCollapsible component above.

File: app/templates/request-code-select-method.html [hunk 3]
```diff
@@ -60,35 +60,6 @@
         'title': question_title,
     }) -%}
 
-        {%- if case_type == 'HH' and request_type != 'individual-code' -%}
-
-            {%- set content -%}
-                {%- autoescape false -%}
-                    <p>{{ _('If you need to answer separately from the people you live with, you can %(open)srequest an individual access code%(close)s to start a separate census.', open='<a href="%s">' % request_individual_code_link, close='</a>')|safe }}</p>
-                {%- endautoescape -%}
-                {%- call onsPanel({
```
Comment (MartinHumphrey) [inline app/templates/request-code-select-method.html:69]: Can remove import of onsPanel component above.

**Review by MartinHumphrey — CHANGES_REQUESTED** [review]
Half a dozen minor comments.

File: app/requests_handlers.py [hunk 4]
```diff
@@ -67,12 +67,17 @@ async def get(self, request):
...
-        request_type = 'individual-code'
-        self.log_entry(request, display_region + '/requests/individual-code')
+        request_type = 'access-code'
+        self.log_entry(request, display_region + '/requests/access-code/individual-information')
+
+        session = await get_session(request)
 
         try:
             if request.cookies.get('RH_SESSION'):
                 session = await get_session(request)
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/requests_handlers.py]: Removed

File: app/templates/request-code-select-method.html (hunk 2 above)
Comment (Paul-Joel) [reply app/templates/request-code-select-method.html:77]: Removed call

File: app/templates/request-code-select-method.html (hunk 3 above)
Comment (Paul-Joel) [reply app/templates/request-code-select-method.html:69]: Removed call

**Review by philwhiles — APPROVED** [review]

File: app/requests_handlers.py [hunk 5]
```diff
@@ -85,12 +90,127 @@ async def post(self, request):
...
+        }
+
+    async def post(self, request):
+        self.setup_request(request)
+        display_region = request.match_info['display_region']
+        request_type = 'paper-form'
+        self.log_entry(request, display_region + '/requests/paper-form/household-information')
+
+        session = await get_session(request)
+        session['attributes']['individual'] = False
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/requests_handlers.py:204]: New ticket raised

File: app/requests_handlers.py [hunk 6]
```diff
@@ -85,12 +90,127 @@ async def post(self, request):
...
+        }
+
+    async def post(self, request):
+        self.setup_request(request)
+        display_region = request.match_info['display_region']
+        request_type = 'access-code'
+        self.log_entry(request, display_region + '/requests/access-code/household-information')
+
+        session = await get_session(request)
+        session['attributes']['individual'] = False
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/requests_handlers.py:167]: New ticket raised

**Review by MartinHumphrey — APPROVED** [review]
Still need timeout of session

**Pull request merged**